sklearn/feature_selection/tests/test_rfe.py

```diff
@@ -108,6 +108,43 @@ def test_rfe():
     assert_array_almost_equal(X_r, X_r_sparse.toarray())
 
 
+def test_rfe_sample_weights():
+    iris = load_iris()
+    X = iris.data
+    y = iris.target
+
+    clf = SVC(kernel="linear")
+    rfe = RFE(estimator=clf, n_features_to_select=1)
+
+    sample_weight_test = 2
+    class_test = 2
+
+    # Case 1 - original dataset
+    rfe.fit(X, y)
+    ranking_original = rfe.ranking_.copy()
+
+    # Case 2 - double the weight of one class's samples
+    w = np.ones(y.shape[0])
+    w[y == class_test] = sample_weight_test
+
+    rfe.fit(X, y, sample_weight=w)
+    ranking_weights = rfe.ranking_.copy()
+
+    # Case 3 - duplicate the samples of one class
+    extra_X = np.tile(X[y == class_test], (sample_weight_test - 1, 1))
+    X_duplicate = np.concatenate((X, extra_X), axis=0)
+
+    n_extra = (y == class_test).sum() * (sample_weight_test - 1)
+    extra_Y = np.ones(n_extra, dtype=int) * class_test
+    y_duplicate = np.concatenate((y, extra_Y), axis=0)
+
+    rfe.fit(X_duplicate, y_duplicate)
+    ranking_duplicate = rfe.ranking_.copy()
+
+    assert_raises(AssertionError, assert_array_equal, ranking_original, ranking_weights)
```

Comment:
We do not use `assert_raises` anymore but with have a context manager in `sklearn.utils._testing.raises`

Reply:
@glemaitre I have just addressed this, thanks for the feedback! But I got curious about why using `_testing.raises` instead of the regular `pytest.raises`. So, after checking the source code, I found this:

> This is only intended to be used in estimator_checks.py where we don't
    want to use pytest. In the rest of the code base, just use pytest.raises
    instead.
    
So, should I just use `pytest.raises`?